exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/interpreter/InterpreterEvaluator.java

```diff
@@ -378,8 +388,14 @@ public ValueHolder visitDoubleConstant(ValueExpressions.DoubleExpression dExpr,
     }
 
     @Override
-    public ValueHolder visitQuotedStringConstant(ValueExpressions.QuotedString e, Integer value) throws RuntimeException {
-      return ValueHolderHelper.getVarCharHolder(getManagedBufferIfAvailable(), e.value);
+    public ValueHolder visitQuotedStringConstant(final ValueExpressions.QuotedString e, Integer value) throws RuntimeException {
+      return getConstantValueHolder(e.value, new Function<DrillBuf, ValueHolder>() {
```

Comment:
@jinfengni, code does not produce the same keys for decimal and string. We pass different functions for each of them.
Example:
1. return ValueHolderHelper.getDecimal38Holder(buffer, decExpr.getBigDecimal().toString()); 
2.  return ValueHolderHelper.getVarCharHolder(buffer, e.value);
Only buffer is the same for string and decimal with the same value but this is acceptable.